File: nma-service/src/main/java/com/devcraftlabs/notification/model/entity/PersonalNotification.java
package com.devcraftlabs.notification.model.entity;

import java.time.Instant;
import java.util.UUID;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Lob;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Table(name = "nma_tbl_personal_notifications")
@Data
public class PersonalNotification {
    public static String PERSONAL_NOTIFICATION_TYPE_DEBUG = "DEBUG";
    public static String PERSONAL_NOTIFICATION_TYPE_MAIL = "MAIL";
    public static String PERSONAL_NOTIFICATION_TYPE_WA = "WHATSAPP";
    public static String PERSONAL_NOTIFICATION_TYPE_SMS = "SMS";

    @Id
    @Column(name = "notification_id", nullable = false)
    private UUID id;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(name = "notification_platform_type", nullable = false, length = 20)
    private String notificationPlatformType;

    @Column(name = "notification_title", nullable = false, length = 60)
    private String title;

    @Column(name = "notification_body", nullable = false, columnDefinition = "TEXT")
    private String body;

    @Column(name = "created_at", nullable = false)
    private Instant createdAt = Instant.now();

    @Column(name = "updated_at")
    private Instant updatedAt;

    @Column(name = "deleted_at")
    private Instant deletedAt;
}

File: sys-config/Model/NMA/PersonalNotification.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Bogus.DataSets;
using SysConfig.Model.NMA.LOV;

namespace SysConfig.Model.NMA;

[Table("nma_tbl_personal_notifications")]
public class PersonalNotification
{
    [Key]
    [Column("notification_id")]
    public Guid NotificationId { get; set; }

    [Required]
    [Column("user_id")]
    public Guid UserId { get; set; }

    [Required]
    [Column("notification_platform_type")]
    [MaxLength(20)]
    public string NotificationPlatformType {get; set;}
    
    [Required]
    [Column("notification_title")]
    [MaxLength(60)]
    public string Title { get; set;}

    [Required]
    [Column("notification_body")]
    [DataType(DataType.Text)]
    public string Body {get;set;}
    
    [Required]
    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    public DateTime? UpdatedAt { get; set; }

    [Column("deleted_at")]
    public DateTime? DeletedAt { get; set; }

    public NotificationPlatformType PlatformType {get; set; }
}
